.github/workflows/publish-packages-main.yaml

```diff
@@ -57,6 +57,59 @@ jobs:
           FILE: ${{ github.workspace }}/packages/${{ github.event.inputs.package }}/package.json
           NEW_VERSION: ${{ github.event.inputs.version }}
 
+      - name: Determine and Set Tags
+        run: |
+          # Find the last two tags that match the package pattern, sort them to get the latest
+          TAGS=$(git tag | grep "^${{ github.event.inputs.package }}-v" | sort -rV | head -n 1)
+          LAST_TAG=$(echo "$TAGS" | sed -n '1p')
+          echo "LAST_TAG=${LAST_TAG}"
+          echo "LAST_TAG=${LAST_TAG}" >> $GITHUB_ENV
+      
+      - name: Generate Changelog
+        id: changelog
+        uses: mikepenz/release-changelog-builder-action@v4
+        with:
+          configurationJson: |
+            {
+              "categories": [
+                  {
+                      "title": "## ${{ github.event.inputs.package }}",
+                      "labels": [
+                          "${{ github.event.inputs.package }}"
+                      ],
+                      "exhaustive": true,
+                      "exhaustive_rules": "false",
+                      "empty_content": "- no matching PRs",
+                      "rules": [
+                          {
+                              "pattern": "${{ github.event.inputs.package }}",
+                              "on_property": "labels"
+                          },
+                          {
+                              "pattern": "merged",
+                              "on_property": "status"
+                          }
+                      ]
+                  }
+              ],
+              "sort": {
+                  "order": "ASC",
+                  "on_property": "mergedAt"
+              },
+              "template": "#{{CHANGELOG}}\n",
+              "pr_template": "- [#{{TITLE}}](#{{URL}})\n",
```

Comment:
Can we change this to match the dolt release notes pattern for PRs? https://github.com/dolthub/dolt/releases/tag/v1.42.1